packages/data/src/plugins/controls.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { controls as dataControls } from '@wordpress/data-controls';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { WC_ADMIN_NAMESPACE } from '../constants';
+
+let optionNames = [];
+const fetches = {};
+
+export const installAndActivatePlugin = ( optionName ) => {
+	return {
+		type: 'INSTALL_AND_ACTIVATE_PLUGIN',
+		optionName,
+	};
+};
+
+export const controls = {
+	...dataControls,
+	INSTALL_AND_ACTIVATE_PLUGIN( { optionName } ) {
+		optionNames.push( optionName );
+
+		return new Promise( ( resolve ) => {
+			setTimeout( function () {
+				const names = optionNames.join( ',' );
+				if ( fetches[ names ] ) {
+					return fetches[ names ].then( ( result ) => {
+						resolve( result[ optionName ] );
+					} );
+				}
+
+				const url = WC_ADMIN_NAMESPACE + '/options?options=' + names;
```

Comment:
This looks like an attempt to batch install/activate, is that right?  I'm a little confused by the endpoint here.

Reply:
Sorry about the confusion here. this file was copy pasta left in the PR by accident when I thought I needed a control earlier on. I'll delete this.